src/core/generators/angular.ts

```diff
@@ -51,14 +52,15 @@ ${
   isRequestOptions && !isGlobalMutator
     ? `type HttpClientOptions = {
   headers?: HttpHeaders | {
-      [header: string]: string | string[];
+    [header: string]: string | string[];
   };
-  observe?: any;
+  context?: HttpContext;
+  observe?: 'body';
   params?: HttpParams | {
-      [param: string]: string | string[];
+    [param: string]: string | number | boolean | ReadonlyArray<string | number | boolean>;
   };
   reportProgress?: boolean;
-  responseType?: any;
+  responseType?: 'json';
```

Comment:
cannot only be 'json' or will limit the user

Reply:
Correct! Sorry, I copy/paste one of the overloads...

I updated it with the real combination of overloads from: https://github.com/angular/angular/blob/c389052bab0db14d4451bb93cf356809a3ad97e3/packages/common/http/src/client.ts#L2552